### PR TITLE
Fix Fastflow ONNX export.

### LINKED ISSUE #1096: [Bug]: Export to ONNX: UnsupportedOperatorError
### Describe the bug

Having set export_mode to openvino or onnx (in other words, enabling export generally) i am receiving an error when it tries to export the model to ONNX format.

### Dataset

Folder

### Model

FastFlow

### Steps to reproduce the behavior

1.Use my config to train
2.Error should appear when training ends

### OS information

OS information:
- OS: Win 10
- Python version: 3.8.16
- Anomalib version: 0.4.0
- PyTorch version: 1.12.1+cu116
- CUDA/cuDNN version: 11.6
- GPU models and configuration: Geforce RTX 3050ti
- Any other relevant information: custom dataset

### Expected behavior

Export fastflow model to ONNX format without error

### Screenshots

_No response_

### Pip/GitHub

pip

### What version/branch did you use?

0.4.0

### Configuration YAML

```yaml
dataset:
  abnormal_dir: bad
  center_crop: null
  extensions: null
  format: folder
  image_size: [224, 224]
  mask: null
  name: vespa
  normal_dir: good
  normal_test_dir: null
  normalization: imagenet
  num_workers: 8
  path: ./../data/images/data_structure_anomalib
  split_ratio: 0.2
  task: classification
  test_batch_size: 8
  test_split_mode: from_dir
  test_split_ratio: 0.2
  tiling:
    apply: false
    random_tile_count: 16
    remove_border_count: 0
    stride: null
    tile_size: null
    use_random_tiling: false
  train_batch_size: 8
  transform_config:
    eval: null
    train: c:\Users\benedict\dev\VespaUseCase\machine_learning\src\d01_config\transform_config_1.yaml
  val_split_mode: from_test
  val_split_ratio: 0.3
  eval_batch_size: 8
  mask_dir: null
  root: ./../data/images/data_structure_anomalib
logging:
  log_graph: false
  logger: tensorboard
metrics:
  image: [F1Score, AUROC]
  pixel: [F1Score, AUROC]
  threshold:
    manual_image: null
    manual_pixel: null
    method: adaptive
model:
  backbone: resnet18
  conv3x3_only: true
  early_stopping:
    metric: image_AUROC
    mode: max
    patience: 20
  flow_steps: 8
  hidden_ratio: 1.0
  lr: 0.001
  name: fastflow
  normalization_method: min_max
  pre_trained: true
  weight_decay: 1e-05
  input_size: [224, 224]
optimization:
  export_mode: onnx
  nncf:
    apply: true
project:
  path: ./../data/results/fastflow/vespa/run
  seed: 42
  unique_dir: false
trainer:
  accelerator: auto
  accumulate_grad_batches: 1
  auto_lr_find: true
  auto_scale_batch_size: false
  benchmark: false
  check_val_every_n_epoch: 1
  default_root_dir: results\fastflow\vespa\run
  detect_anomaly: false
  deterministic: false
  devices: 1
  enable_checkpointing: true
  enable_model_summary: true
  enable_progress_bar: true
  fast_dev_run: false
  gradient_clip_algorithm: norm
  gradient_clip_val: 0
  limit_predict_batches: 1.0
  limit_test_batches: 1.0
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  log_every_n_steps: 5
  max_epochs: 130
  max_steps: -1
  max_time: null
  min_epochs: null
  min_steps: null
  move_metrics_to_cpu: false
  multiple_trainloader_mode: max_size_cycle
  num_nodes: 1
  num_sanity_val_steps: 0
  overfit_batches: 0.0
  plugins: null
  precision: 32
  profiler: null
  reload_dataloaders_every_n_epochs: 0
  replace_sampler_ddp: true
```

### Logs

```shell
---------------------------------------------------------------------------
UnsupportedOperatorError                  Traceback (most recent call last)
Cell In[16], line 1
----> 1 trainer.fit(model=model, datamodule=datamodule)

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\trainer.py:608, in Trainer.fit(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    606 model = self._maybe_unwrap_optimized(model)
    607 self.strategy._lightning_module = model
--> 608 call._call_and_handle_interrupt(
    609     self, self._fit_impl, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path
    610 )

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\call.py:38, in _call_and_handle_interrupt(trainer, trainer_fn, *args, **kwargs)
     36         return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
     37     else:
---> 38         return trainer_fn(*args, **kwargs)
     40 except _TunerExitException:
     41     trainer._call_teardown_hook()

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\trainer.py:650, in Trainer._fit_impl(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    643 ckpt_path = ckpt_path or self.resume_from_checkpoint
    644 self._ckpt_path = self._checkpoint_connector._set_ckpt_path(
    645     self.state.fn,
    646     ckpt_path,  # type: ignore[arg-type]
    647     model_provided=True,
    648     model_connected=self.lightning_module is not None,
    649 )
--> 650 self._run(model, ckpt_path=self.ckpt_path)
    652 assert self.state.stopped
    653 self.training = False

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\trainer.py:1112, in Trainer._run(self, model, ckpt_path)
   1108 self._checkpoint_connector.restore_training_state()
   1110 self._checkpoint_connector.resume_end()
-> 1112 results = self._run_stage()
   1114 log.detail(f"{self.__class__.__name__}: trainer tearing down")
   1115 self._teardown()

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\trainer.py:1191, in Trainer._run_stage(self)
   1189 if self.predicting:
   1190     return self._run_predict()
-> 1191 self._run_train()

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\trainer.py:1214, in Trainer._run_train(self)
   1211 self.fit_loop.trainer = self
   1213 with torch.autograd.set_detect_anomaly(self._detect_anomaly):
-> 1214     self.fit_loop.run()

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\loops\loop.py:206, in Loop.run(self, *args, **kwargs)
    203         break
    204 self._restarting = False
--> 206 output = self.on_run_end()
    207 return output

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\loops\fit_loop.py:323, in FitLoop.on_run_end(self)
    320 log.detail(f"{self.__class__.__name__}: train run ended")
    322 # hook
--> 323 self.trainer._call_callback_hooks("on_train_end")
    324 self.trainer._call_lightning_module_hook("on_train_end")
    325 self.trainer._call_strategy_hook("on_train_end")

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\trainer.py:1394, in Trainer._call_callback_hooks(self, hook_name, *args, **kwargs)
   1392     if callable(fn):
   1393         with self.profiler.profile(f"[Callback]{callback.state_key}.{hook_name}"):
-> 1394             fn(self, self.lightning_module, *args, **kwargs)
   1396 if pl_module:
   1397     # restore current_fx when nested context
   1398     pl_module._current_fx_name = prev_fx_name

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\anomalib\utils\callbacks\export.py:47, in ExportCallback.on_train_end(***failed resolving arguments***)
     45 logger.info("Exporting the model")
     46 Path(self.dirpath).mkdir(parents=True, exist_ok=True)
---> 47 export(
     48     model=pl_module,
     49     input_size=self.input_size,
     50     export_root=self.dirpath,
     51     export_mode=self.export_mode,
     52 )

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\anomalib\deploy\export.py:80, in export(model, input_size, export_mode, export_root)
     77             meta_data[key] = value.numpy().tolist()
     78     json.dump(meta_data, metadata_file, ensure_ascii=False, indent=4)
---> 80 onnx_path = _export_to_onnx(model, input_size, export_path)
     81 if export_mode == ExportMode.OPENVINO:
     82     _export_to_openvino(export_path, onnx_path)

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\anomalib\deploy\export.py:97, in _export_to_onnx(model, input_size, export_path)
     86 """Export model to onnx.
     87 
     88 Args:
   (...)
     94     Path: Path to the exported onnx model.
     95 """
     96 onnx_path = export_path / "model.onnx"
---> 97 torch.onnx.export(
     98     model.model,
     99     torch.zeros((1, 3, *input_size)).to(model.device),
    100     onnx_path,
    101     opset_version=11,
    102     input_names=["input"],
    103     output_names=["output"],
    104 )
    106 return onnx_path

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\torch\onnx\__init__.py:350, in export(model, args, f, export_params, verbose, training, input_names, output_names, operator_export_type, opset_version, do_constant_folding, dynamic_axes, keep_initializers_as_inputs, custom_opsets, export_modules_as_functions)
     74 r"""
     75 Exports a model into ONNX format. If ``model`` is not a
     76 :class:`torch.jit.ScriptModule` nor a :class:`torch.jit.ScriptFunction`, this runs
   (...)
    345     model to the file ``f`` even if this is raised.
    346 """
    348 from torch.onnx import utils
--> 350 return utils.export(
    351     model,
    352     args,
    353     f,
    354     export_params,
    355     verbose,
    356     training,
    357     input_names,
    358     output_names,
    359     operator_export_type,
    360     opset_version,
    361     do_constant_folding,
    362     dynamic_axes,
    363     keep_initializers_as_inputs,
    364     custom_opsets,
    365     export_modules_as_functions,
    366 )

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\torch\onnx\utils.py:163, in export(model, args, f, export_params, verbose, training, input_names, output_names, operator_export_type, opset_version, do_constant_folding, dynamic_axes, keep_initializers_as_inputs, custom_opsets, export_modules_as_functions)
    145 def export(
    146     model,
    147     args,
   (...)
    160     export_modules_as_functions=False,
    161 ):
--> 163     _export(
    164         model,
    165         args,
    166         f,
    167         export_params,
    168         verbose,
    169         training,
    170         input_names,
    171         output_names,
    172         operator_export_type=operator_export_type,
    173         opset_version=opset_version,
    174         do_constant_folding=do_constant_folding,
    175         dynamic_axes=dynamic_axes,
    176         keep_initializers_as_inputs=keep_initializers_as_inputs,
    177         custom_opsets=custom_opsets,
    178         export_modules_as_functions=export_modules_as_functions,
    179     )

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\torch\onnx\utils.py:1074, in _export(model, args, f, export_params, verbose, training, input_names, output_names, operator_export_type, export_type, opset_version, do_constant_folding, dynamic_axes, keep_initializers_as_inputs, fixed_batch_size, custom_opsets, add_node_names, onnx_shape_inference, export_modules_as_functions)
   1071     dynamic_axes = {}
   1072 _validate_dynamic_axes(dynamic_axes, model, input_names, output_names)
-> 1074 graph, params_dict, torch_out = _model_to_graph(
   1075     model,
   1076     args,
   1077     verbose,
   1078     input_names,
   1079     output_names,
   1080     operator_export_type,
   1081     val_do_constant_folding,
   1082     fixed_batch_size=fixed_batch_size,
   1083     training=training,
   1084     dynamic_axes=dynamic_axes,
   1085 )
   1087 # TODO: Don't allocate a in-memory string for the protobuf
   1088 defer_weight_export = (
   1089     export_type is not torch.onnx.ExportTypes.PROTOBUF_FILE
   1090 )

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\torch\onnx\utils.py:731, in _model_to_graph(model, args, verbose, input_names, output_names, operator_export_type, do_constant_folding, _disable_torch_constant_prop, fixed_batch_size, training, dynamic_axes)
    728 params_dict = _get_named_param_dict(graph, params)
    730 try:
--> 731     graph = _optimize_graph(
    732         graph,
    733         operator_export_type,
    734         _disable_torch_constant_prop=_disable_torch_constant_prop,
    735         fixed_batch_size=fixed_batch_size,
    736         params_dict=params_dict,
    737         dynamic_axes=dynamic_axes,
    738         input_names=input_names,
    739         module=module,
    740     )
    741 except Exception as e:
    742     torch.onnx.log("Torch IR graph at exception: ", graph)

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\torch\onnx\utils.py:308, in _optimize_graph(graph, operator_export_type, _disable_torch_constant_prop, fixed_batch_size, params_dict, dynamic_axes, input_names, module)
    306     _C._jit_pass_onnx_set_dynamic_input_shape(graph, dynamic_axes, input_names)
    307 _C._jit_pass_onnx_lint(graph)
--> 308 graph = _C._jit_pass_onnx(graph, operator_export_type)
    309 _C._jit_pass_onnx_lint(graph)
    310 _C._jit_pass_lint(graph)

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\torch\onnx\__init__.py:416, in _run_symbolic_function(*args, **kwargs)
    413 def _run_symbolic_function(*args, **kwargs):
    414     from torch.onnx import utils
--> 416     return utils._run_symbolic_function(*args, **kwargs)

File c:\Users\benedict\anaconda3\envs\anomalib_env\lib\site-packages\torch\onnx\utils.py:1421, in _run_symbolic_function(g, block, n, inputs, env, operator_export_type)
   1417         return g.at(  # type: ignore[attr-defined]
   1418             op_name, *inputs, overload_name=_get_aten_op_overload_name(n), **attrs
   1419         )
   1420     else:
-> 1421         raise symbolic_registry.UnsupportedOperatorError(
   1422             domain, op_name, opset_version
   1423         )
   1424 except RuntimeError:
   1425     if operator_export_type == _C_onnx.OperatorExportTypes.ONNX_FALLTHROUGH:

UnsupportedOperatorError: Exporting the operator ::_convolution_mode to ONNX opset version 11 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jasonvanzelm**
# Description

- Provide a summary of the modification as well as the issue that has been resolved. List any dependencies that this modification necessitates.

- Fixes #1096 

## Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

**Comment by blaz-r** [comment]
Looks good. From what I tested it works ok, but at some point there is some difference on 7th decimal which probably won't cause problems. According to this https://github.com/pytorch/pytorch/pull/89107 "same" will be supported in PyTorch 2.1, but in the meantime this workaround look great.

**Comment by samet-akcay** [comment]
> Looks good. From what I tested it works ok, but at some point there is some difference on 7th decimal which probably won't cause problems. According to this [pytorch/pytorch#89107](https://github.com/pytorch/pytorch/pull/89107) "same" will be supported in PyTorch 2.1, but in the meantime this workaround look great.

Looks like PyTorch v2.1 will be released some time in october. Meanwhile, we could use this approach to tackle this issue.

**Review by samet-akcay — APPROVED** [review]
@jasonvanzelm, thanks for your contribution!

File: src/anomalib/models/fastflow/torch_model.py [hunk 1]
```diff
@@ -41,10 +41,14 @@ def subnet_conv_func(kernel_size: int, hidden_ratio: float) -> Callable:
 
     def subnet_conv(in_channels: int, out_channels: int) -> nn.Sequential:
         hidden_channels = int(in_channels * hidden_ratio)
+        # setting padding="same" in nn.Conv2d breaks the onnx export so we pad by hand
```
Comment (samet-akcay) [inline src/anomalib/models/fastflow/torch_model.py:44]:
```suggestion
        # NOTE: setting padding="same" in nn.Conv2d breaks the onnx export so manual padding required.
        # TODO: Use padding="same" in nn.Conv2d once PyTorch v2.1 is released
```

**Review by samet-akcay — CHANGES_REQUESTED** [review]
sorry, just noticed a minor issue here

**Pull request merged**